### ISSUE TITLE
Core Crossplane upgrade tests

### ISSUE THREAD

**Issue by hasheddan** [crossplane/test#4]
We should be running periodic upgrade tests of supported Crossplane versions to the next version (including latest builds). For example, right now we should be running:
- Upgrade from v1.0.0 to latest build on `master`
- Upgrade from v1.0.0 to latest build on `release-1.0`

**Comment by rahulgrover99** [comment]
@hasheddan Just wanted to check with you once regarding the steps to upgrade crossplane from stable to master once. Please let me know if there's something that might be missing. 
```
# To install the stable version of crossplane. 

kubectl create namespace crossplane-system
helm repo add crossplane-stable https://charts.crossplane.io/stable
helm repo update
helm install crossplane --namespace crossplane-system crossplane-stable/crossplane
```

Then we could run the e2e tests and then upgrade to master. 
```
helm repo add crossplane-master https://charts.crossplane.io/master/
helm repo update
helm upgrade crossplane --namespace crossplane-system crossplane-master/crossplane
```
Then we could just run the tests again.

Also, I see many charts in [Helm Crossplane Master](https://charts.crossplane.io/master/). So when we run `upgrade crossplane`, does it pick the latest chart? If yes, then how can we choose a specific version from the charts available? I am guessing it will be using a `--version` tag?

**Comment by rahulgrover99** [comment]
Having a look at [docs](https://crossplane.io/docs/v1.1/guides/upgrading-to-v1.x.html), I think I might have missed updating the CRDs since Helm won't take care of that.

**Comment by hasheddan** [comment]
Fixed by #10
